sentry-android-core/src/main/java/io/sentry/android/core/internal/gestures/SentryGestureListener.java

```diff
@@ -178,12 +191,128 @@ private void addBreadcrumb(
 
     hub.addBreadcrumb(
         Breadcrumb.userInteraction(
-            eventType, ViewUtils.getResourceId(target), className, additionalData),
+            eventType, ViewUtils.getResourceIdWithFallback(target), className, additionalData),
         hintMap);
   }
 
+  private void startTracing(final @NotNull View target, final @NotNull String eventType) {
+    if (!(options.isTracingEnabled() && options.isEnableUserInteractionTracing())) {
+      return;
+    }
+
+    final Activity activity = activityRef.get();
+    if (activity == null) {
+      options.getLogger().log(SentryLevel.DEBUG, "Activity is null, no transaction captured.");
+      return;
+    }
+
+    final String viewId;
+    try {
+      viewId = ViewUtils.getResourceId(target);
+    } catch (Resources.NotFoundException e) {
+      options
+          .getLogger()
+          .log(
+              SentryLevel.DEBUG,
+              "View id cannot be retrieved from Resources, no transaction captured.");
+      return;
+    }
+
+    final View view = activeView.get();
+    if (activeTransaction != null) {
+      if (target.equals(view)
+          && eventType.equals(activeEventType)
+          && !activeTransaction.isFinished()) {
+        options
+            .getLogger()
+            .log(
+                SentryLevel.DEBUG,
+                "The view with id: "
+                    + viewId
+                    + " already has an ongoing transaction assigned. Rescheduling finish");
+
+        final Long idleTimeout = options.getIdleTimeout();
+        if (idleTimeout != null) {
+          // reschedule the finish task for the idle transaction, so it keeps running for the same
+          // view
+          activeTransaction.scheduleFinish(idleTimeout);
+        }
+        return;
+      } else {
+        // as we allow a single UI transaction running on the bound Scope, we finish the previous
+        // one, if it's a new view
+        activeTransaction.finish();
+        activeTransaction = null;
```

Comment:
Here maybe, we'd need to remove the transaction from scope, in case `activeTransaction` has an unfinished child, `finish` wouldn't do anything, hence the new transaction wouldn't be added to Scope, since there's still an ongoing one.